x-pack/plugins/ingest_manager/server/services/epm/kibana/index_pattern/install.ts

```diff
@@ -86,6 +87,17 @@ export async function installIndexPatterns(
     savedObjectsClient,
     InstallationStatus.installed
   );
+
+  // TODO: move to install package
+  // cache all installed packages if they don't exist
+  const packagesPromise = installedPackages.map((pkg) => {
+    const pkgkey = getCacheKey(pkg.pkgName + '-' + pkg.pkgVersion);
+    if (!cacheHas(pkgkey)) {
+      return Registry.getArchiveInfo(pkg.pkgName, pkg.pkgVersion);
+    }
+  });
+  await Promise.all(packagesPromise);
```

Comment:
i think i can remove this in another optimization, instead of creating the single index pattern for all datasets of all installed packages, getting the existing pattern from kibana and deduping it and merging it with the new one, possibly.  this will remove the need to get the archive info for all installed packages on an install.